src/tvm_wrapper.cu

```diff
@@ -97,10 +75,11 @@ cudaError_t _SinglePrefillWithKVCacheNoLSE(DTypeIn* q, DTypeIn* k, DTypeIn* v, D
   return cudaSuccess;
 }
 
-int _FlashInferSinglePrefillWithKVCache(DLTensor* q, DLTensor* k, DLTensor* v, bool causal,
-                                        int64_t qkv_layout, int64_t rotary_mode,
+int _FlashInferSinglePrefillWithKVCache(DLTensor* q, DLTensor* k, DLTensor* v, DLTensor* tmp,
+                                        bool causal, int64_t qkv_layout, int64_t rotary_mode,
                                         bool allow_fp16_qk_reduction, double rope_scale,
                                         double rope_theta, DLTensor* o) {
+  // `tmp` is a 4096 * 4096 float16 scratch space.
```

Comment:
```suggestion
  // `tmp` is a user provided scratch space.
```
`tmp` is used as a float32 tensor, and the required space is hardware dependent, 16mb is a suggested value.

Reply:
I see. Will use fp32 instead of fp16

Reply:
updated